parachain/runtime/litentry/src/migration/mod.rs

```diff
@@ -1 +1,362 @@
+// democracy:                      -                     0
+// bounties:                       -                     0
+// xcmpQueue:                      -                     3
+// polkadotXcm:                    -                     0
+// developerCommittee              -                     0
+// developerCommitteeMembership    -                     0
+// transactionPayment:             V1Ancient             0
+// vesting:                        V1                    0
+
+// Our target         storage version:      pallet version: (stable2407)
+// scheduler:                      -                     0 => 4
+// multisig:                       -                     0 => 1
+// preimage:                       -                     0 => 1
+// balances:                       -                     0 => 1
+// democracy:                      -                     0 => 1
+// bounties:                       -                     0 => 4
+// xcmpQueue:                      -                     3 => 5
+// polkadotXcm:                    -                     0 => 1
+// developerCommittee              -                     0 => 4
+// developerCommitteeMembership    -                     0 => 4
+// transactionPayment:             V1Ancient => V2       0   (it is built by genesis, so maybe no fix is fine)
+// vesting:                        V1                    0
+
+// In try-runtime, current implementation, the storage version is not checked,
+// Pallet version is used instead.
+use frame_support::traits::{
+	Get, GetStorageVersion, OnRuntimeUpgrade, PalletInfoAccess, StorageVersion,
+};
+use frame_system::pallet_prelude::BlockNumberFor;
+use pallet_balances::InactiveIssuance;
+use pallet_scheduler::Agenda;
+use sp_std::marker::PhantomData;
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+pub type Migrations<Runtime> = (
+	// Scheduler V0 => V4
+	// The official pallet does not provide any available migration
+	// We, Litentry Storage have two old unexecuted expired root tasks.
+	// This storage should be clean up and update storage version to V4 directly.
+	// PS: Looks like two old tasks fits V2/V3 structure
+	RemoveSchedulerOldStorage<Runtime>,
+	// Multsig V0 => V1
+	// Remove old unsettled call storage and do the refund
+	// Does not matter if we do have old storage or not
+	// Should simply works
+	pallet_multisig::migrations::v1::MigrateToV1<Runtime>,
+	// Preimage V0 => V1
+	// Migrate old StatusFor and PreimageFor Storage into new Storage
+	pallet_preimage::migration::v1::Migration<Runtime>,
+	// Balances V0 => V1
+	// The official pallet migration is not need since we do not have any XCM deactive accounts
+	// But our onchain inactiveIssuance storage of pallet_balance is non-negative
+	// TODO: Where does this number come from?
+	BalancesUpdateStorageVersionResetInactive<Runtime>,
+	// Democracy V0 => V1
+	// This migration only effects onging proposal/referedum, NextExternal
```

Comment:
Does it mean it's better to wait until there's no on-going ref/proposals to do this migration?